api/service/syncing/syncing.go

```diff
@@ -770,28 +784,60 @@ Loop:
 			currentHeight := bc.CurrentBlock().NumberU64()
 
 			if currentHeight >= otherHeight {
-				utils.Logger().Info().
-					Msgf("[SYNC] Node is now IN SYNC! (isBeacon: %t, ShardID: %d, otherHeight: %d, currentHeight: %d)",
-						isBeacon, bc.ShardID(), otherHeight, currentHeight)
+				utils.Logger().Debug().
+					Bool("isBeacon", isBeacon).
+					Uint32("ShardID", bc.ShardID()).
+					Uint64("otherHeight", otherHeight).
+					Uint64("currentHeight", currentHeight).
+					Msg("[SYNC] Node is now IN SYNC!")
 				break Loop
 			} else {
 				utils.Logger().Debug().
-					Msgf("[SYNC] Node is Not in Sync (isBeacon: %t, ShardID: %d, otherHeight: %d, currentHeight: %d)",
-						isBeacon, bc.ShardID(), otherHeight, currentHeight)
+					Bool("isBeacon", isBeacon).
+					Uint32("ShardID", bc.ShardID()).
+					Uint64("otherHeight", otherHeight).
+					Uint64("currentHeight", currentHeight).
+					Msg("[SYNC] Node is Not in Sync.")
 			}
 			startHash := bc.CurrentBlock().Hash()
 			size := uint32(otherHeight - currentHeight)
 			if size > SyncLoopBatchSize {
 				size = SyncLoopBatchSize
 			}
-			err := ss.ProcessStateSync(startHash[:], size, bc, worker)
-			if err != nil {
-				utils.Logger().Error().Err(err).
-					Msgf("[SYNC] ProcessStateSync failed (isBeacon: %t, ShardID: %d, otherHeight: %d, currentHeight: %d)",
-						isBeacon, bc.ShardID(), otherHeight, currentHeight)
-				// should we still call UpdateConsensusInformation() upon state sync failure?
-				// how to handle error here?
+
+			retryCount := 0
+			var err error
+			for {
```

Comment:
if you know the max amount of attempts, then perhaps better to not use the infinite loop and just a plain for loop?

Reply:
It was cleaner this way so that logging (line 816-821) can be done for the case using up all the retry limits.